pkg/channel/external/rwrecorder.go

```diff
@@ -0,0 +1,111 @@
+package external
+
+/*!
+Adapted from the Go 1.19.2 source code
```

Comment:
Please comment changes here in case we want to upgrade it.

Reply:
Removed duplicate file.